core/models.py

```diff
@@ -120,6 +120,13 @@ def vote(self: "Post", user: User, choice: str) -> None:
     def get_images(self: "Post") -> models.QuerySet:
         return Image.objects.filter(post=self)
 
+    def save_post(self: "Post", user: User) -> "SavedPost":
+        saved_post, created = SavedPost.objects.get_or_create(user=user, post=self)
+        return saved_post
+
+    def remove_saved_post(self: "Post", user: User) -> None:

```

Comment:
Architecture wise it would be better to have this method on SavedPost model. This way post model does not know anything about SavedPost, which is good cause we can easily remove savedPost or copy post model to new project. 
So please move this method into SavedPost